miden-tx/src/host/account_delta_tracker.rs

```diff
@@ -201,19 +201,13 @@ impl AccountVaultDeltaTracker {
         for (faucet_id, amount) in self.fungible_assets {
             if amount > 0 {
                 added_assets.push(Asset::Fungible(
-                    FungibleAsset::new(
-                        AccountId::new_unchecked(faucet_id.into()),
```

Comment:
I removed the `unchecked` API, because there are `unwrap`s and `assert`s in the rest of the code that can be hit by these.